src/dune_rules/findlib.ml

```diff
@@ -206,10 +206,13 @@ let to_dune_library (t : Findlib.Package.t) ~dir_contents ~ext_lib ~external_loc
                     | Error e -> Error e))))
     in
     let modules = Lib_info.Source.External None in
+    let name = t.name in
+    let library_id = Lib_info.Library_id.external_ ~loc ~src_dir name in
```

Comment:
I think difference for id's of external vs local libraries is that external id's really are identified by their name. That's because we never keep searching for external libraries after we find the first name. So I think the id type should be more like:

```
type t =
  | External of Lib_name.t
  | Local of what_is_library_id_now
```

Reply:
In fact, something like:

```
$ cat lib_id.mli
module Local : sig
  type t
end
type t = External of Lib_name.t | Local of Local.t
```
And you can keep using `Lib_id.Local.t` in places like `Ml_sources` where it only makes sense to talk about local libraries.